app/src/main/java/org/mozilla/fenix/onboarding/view/OnboardingPage.kt

```diff
@@ -0,0 +1,165 @@
+import org.mozilla.fenix.compose.button.SecondaryButton
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * A composable for displaying onboarding page content.
+ *
+ * @param pageState [OnboardingPageState] The page content that's displayed.
+ * @param onDismiss Invoked when the user clicks the close button.
+ * @param onPrimaryButtonClick Invoked when the user clicks the primary button.
+ * @param onSecondaryButtonClick Invoked when the user clicks the secondary button.
+ * @param modifier The modifier to be applied to the Composable.
+ */
+@Composable
+fun OnboardingPage(
+    pageState: OnboardingPageState,
+    onDismiss: () -> Unit,
+    onPrimaryButtonClick: () -> Unit,
+    onSecondaryButtonClick: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    BoxWithConstraints(
+        modifier = Modifier
+            .background(FirefoxTheme.colors.layer1)
+            .padding(bottom = 16.dp)
+            .then(modifier),
+    ) {
+        val boxWithConstraintsScope = this
+        Column(
+            modifier = Modifier
+                .fillMaxSize()
+                .verticalScroll(rememberScrollState()),
+            horizontalAlignment = Alignment.CenterHorizontally,
+            verticalArrangement = Arrangement.SpaceBetween,
+        ) {
+            IconButton(
+                onClick = onDismiss,
+                modifier = Modifier.align(Alignment.End),
+            ) {
+                Icon(
+                    painter = painterResource(id = R.drawable.mozac_ic_close),
+                    contentDescription = stringResource(R.string.content_description_close_button),
+                    tint = FirefoxTheme.colors.iconPrimary,
+                )
+            }
+
+            Column(
+                modifier = Modifier.padding(horizontal = 16.dp),
+                horizontalAlignment = Alignment.CenterHorizontally,
+                verticalArrangement = Arrangement.Center,
+            ) {
+                Image(
+                    painter = painterResource(id = pageState.image),
+                    contentDescription = null,
+                    modifier = Modifier
+                        .height(boxWithConstraintsScope.maxHeight.times(IMAGE_HEIGHT_RATIO)),
+                )
+
+                Spacer(modifier = Modifier.height(32.dp))
+
+                Text(
+                    text = pageState.title,
+                    color = FirefoxTheme.colors.textPrimary,
+                    textAlign = TextAlign.Center,
+                    style = FirefoxTheme.typography.headline5,
+                )
+
+                Spacer(modifier = Modifier.height(16.dp))
+
+                Text(
+                    text = pageState.description,
+                    color = FirefoxTheme.colors.textSecondary,
+                    textAlign = TextAlign.Center,
+                    style = FirefoxTheme.typography.body2,
+                )
+            }
+
+            Spacer(modifier = Modifier.height(16.dp))
```

Comment:
Does this `Spacer` act like item in the this Column with SpaceBetween? Instead of having this Spacer here, could we just carryover the 32dp vertical padding in the Column on line 78 https://searchfox.org/mozilla-mobile/source/fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/Onboarding.kt#167?

Reply:
Good catch, fixed now 👍